tests/statictypes/t19819.nim

```diff
@@ -0,0 +1,7 @@
+discard """
+  errormsg: "expression 'E.N' type cannot be inferred"
+"""
+
+type
+  Example[N: static int] = distinct void
+  What[E: Example] = Example[E.N + E.N]
```

Comment:
How does this work?  semTemplateExpr is about templates and the example doesn't contain any.